File: vue/src/services/PrizeService.js
import axios from 'axios';


export default {
    list() {
        return axios.get('/prizes')
    },
    listAvailablePrizes(familyId) {
        return axios.get('/available_prizes', { params: {familyId}})
    },
    createPrize(prize, family_id) {
        return axios.post('/create_prize', prize, { params: {family_id}});
    }
}

File: java/src/main/java/com/techelevator/Service/PrizeService.java
package com.techelevator.Service;

import com.techelevator.dao.PrizeDao;
import com.techelevator.model.Prize;
import org.springframework.stereotype.Component;

import java.util.List;

@Component
public class PrizeService {
    private PrizeDao prizeDao;
    public PrizeService(PrizeDao prizeDao) {
        this.prizeDao = prizeDao;
    }

    public List<Prize> findAll() {
        return prizeDao.findAll();
    }
    public List<Prize> listAvailablePrizes(int familyId) {
        return prizeDao.listAvailablePrizes(familyId);
    }
    public int createPrize(Prize prize) {
        return prizeDao.createPrize(prize);
    }
}

